instat/ucrCore.vb

```diff
@@ -280,7 +280,7 @@ Public Class ucrCore
                 If bAdd Then
                     lstAllRCodes(i).AddParameter(lstAllRParameters(i))
                 Else
-                    lstAllRCodes(i).RemoveParameter(lstAllRParameters(i))
+                    lstAllRCodes(i).RemoveParameterByName(lstAllRParameters(i).strArgumentName)

```

Comment:
I have changed the core control here which should fix the bug with weights on your dialog.